src/crypto/password/pbkdf2.clj

```diff
@@ -50,7 +61,13 @@
   crypto.password.pbkdf2/encrypt function. Returns true the string match, false
   otherwise."
   [raw encrypted]
-  (let [[i s _]    (str/split encrypted #"\$")
-        salt       (decode-str s)
-        iterations (decode-int i)]
-    (crypto/eq? encrypted (encrypt raw iterations salt))))
+   (let [[i s a h] (str/split encrypted #"\$")
+         algorithm (if (nil? h) "HMAC-SHA1" (apply str (map char (decode-str a))))
+         ; passwords in earlier versions of this library contained 3 parts,
+         ; need to convert to 4 part format for comparison
```

Comment:
I don't think we should do this. As I mentioned in my previous comment, I think that we should still produce a 3-part output for HMAC-SHA1, and 4-port for everything else. I don't think there's a compelling reason to do otherwise aside from aesthetics.